lib/iris/cube.py

```diff
@@ -165,19 +165,25 @@ class CubeList(list):
 
     """
 
-    def __new__(cls, list_of_cubes=None):
-        """Given a :class:`list` of cubes, return a CubeList instance."""
-        cube_list = list.__new__(cls, list_of_cubes)
-
-        # Check that all items in the incoming list are cubes. Note that this
-        # checking does not guarantee that a CubeList instance *always* has
-        # just cubes in its list as the append & __getitem__ methods have not
-        # been overridden.
-        if not all([isinstance(cube, Cube) for cube in cube_list]):
+    def __init__(self, iterable_of_cubes=None):
+        """Given an iterable of cubes, return a CubeList instance."""
+        if iterable_of_cubes is None:
+            iterable_of_cubes = []
+        # Run the iterable, and capture as a list.
+        list_of_cubes = list(iterable_of_cubes)
+
+        # Check that all items in the incoming list (if any) are cubes.
+        # Note that this checking does not (yet) guarantee that a CubeList
+        # instance *always* contains only cubes, as the append & __setitem__ methods
+        # (for example), have not  been overridden.
+        if not all(isinstance(cube, Cube) for cube in list_of_cubes):
```

Comment:
There was a lot of discussion at #3238 about allowing cube-like objects into the cubelist.  Probably we need to come to a conclusion about that before merging this.  Or just remove all checks, since Iris has effectively been without them up to now.

🦆🦆🦆

Reply:
We've discussed this in a @SciTools/peloton meeting and the consensus is that this can proceed without #3238

Reply:
Sorry, I think my above comment was ambiguous: I meant to say we need a decision about whether we want to allow cube-like (duck typed) objects into cubelists.  This line would obviously disallow them to be included at  instantiation.  Within #3238 both @pp-mo and @pelson said they wanted to allow duck-cubes into cubelists.

Reply:
Well we've moved on a bit now , especially with [UCube coming](https://github.com/SciTools-incubator/iris-ugrid/blob/master/iris_ugrid/ucube.py#L13) 
( though that isn't fully decided + agreed as a final approach yet ).

So I think I'd be happy to broaden this to a duck test.
But we need to think about what.

The previously suggested `hasattr(cube, 'add_aux_coord')` is a bit specific, but I guess only one test is really quick.
Or we could aim to accept an array-like object with cube elements and phenomenon identity.
Maybe something like `hasattr(cube, 'shape') and hasattr(cube, 'standard_name')` ?